
// 注册流程：
// 1 [Auth] 发送 m_1001_tos注册账号
//
// 登陆流程：
// 1 [Auth] 发送 m_1002_tos协议登陆,并且获得登陆验证密钥
// 2 [Gate] 发送 m_2001_tos进入游戏, 并且获得玩家各种信息
//
// 重连流程
// 1 重复 登陆流程 即可


// 账号注册
message m_1001_tos {
    required string user_name = 1;  // 用户名
    required string password = 2;   // 密码
}

message m_1001_toc {
    required string user_name = 1;  // 用户名
    required string password = 2;   // 密码
    required uint32 account_id = 3; // 账号ID
}

// 玩家本地登陆
message m_1002_tos {
    required string user_name = 1;  // 用户名
    required string password = 2;   // 密码
}

message m_1002_toc {
    required uint32 time = 1;       // 服务器时间戳
    required uint32 account_id = 2; // 账号ID
    required string verify_key = 3; // 登陆验证密钥
}

// 玩家渠道登陆
message m_1003_tos {
    required string user_name = 1;  // 用户名
    required uint32 channel_id = 2; // 渠道编号 1 微信
    required string uuid = 3;       // 渠道唯一标识符, 比如微信的openid
    optional string name = 4;       // 玩家名
    optional string head_frame = 5; // 头像框
    optional string head_icon = 6;  // 头像
    optional uint32 sex = 7;        // 性别
}

message m_1003_toc {
    required uint32 time = 1;       // 服务器时间戳
    required uint32 account_id = 2; // 账号ID
    required string verify_key = 3; // 登陆验证密钥
}


// 玩家进入游戏
message m_2001_tos {
    required uint32 account_id = 1;    // 账号ID
    required string verify_key = 2;    // 登陆验证密钥
}

message m_2001_toc {
    required p_user_info user_info = 1; // 玩家信息
    repeated p_room_info room_info = 2; // 房间信息
    optional p_game_info game_info = 3; // 游戏信息
}

message p_user_info {
    required uint32 account_id = 1;     // 账号ID, 客户端用于显示的ID
    required string uuid = 2;           // 渠道唯一标识符, 比如微信的openid
    optional string name = 3;           // 玩家名
    optional string head_frame = 4;     // 头像框
    optional string head_icon = 5;      // 头像
    optional uint32 sex = 6;            // 性别
    optional uint32 gold = 7;           // 金币
    optional uint32 point = 8;          // 总积分
    optional uint32 room_id = 9;        // 之前的房间ID, 如果之前是手动退出房间, 则此处为0
    optional uint32 room_type = 10;      // 之前的房间类型, 如果之前是手动退出房间, 则此处为0
    optional uint32 proxy_id = 11;      // 上级代理ID
}

message p_room_info {
    required uint32 room_type = 1;      // 房间类型: 1 跑的快，2 转转麻将。
    required uint32 room_id = 2;        // 拥有的该类型的房间ID
    repeated p_room_price room_price = 3;  // 房间创建费用
}

message p_room_price {
    required uint32 rounds = 1;             // 回合数
    required uint32 gold_price = 2;         // 需要的金币数量
}

message p_game_info {
    optional string contact = 1;            // 联系我们
    repeated p_recharge_info recharges = 2; // 充值信息
    optional uint32 poker_per_price = 3;    // 扑克每局费用
    optional uint32 mahjong_per_price = 4;  // 麻将每局费用
}

message p_recharge_info {
    required uint32 money = 1;              // 充值的人民币(分)数量
    required uint32 ingot = 2;              // 获取的代币数量
}

// 绑定代理
message m_2002_tos {
    required uint32 proxy_id = 1;    // 代理账号ID
}

message m_2002_toc {
    required uint32 proxy_id = 1;   // 代理账号ID
}
